lib/NodeDomain.js

```diff
@@ -138,14 +143,27 @@ define(function (require, exports, module) {
     NodeDomain.prototype._load = function () {
         if (!this._domainPath) {
             // if path not provided, validate that the domain is already loaded into the connection
-            if (!this.connection.domains[this._domainName]) {
-                var reason = "Domain '" + this._domainName + "' does not exist, nor was a domain path provided";
-                return Promise.reject(reason);
-            } else {
+            // If it isn't immediately there, wait a bit before trying again
+            // Because the websocket starts accepting requests before the domain may have been fully loaded
+            var domainPromise = this.connection.domains[this._domainName] ?
+                Promise.resolve().bind(this) :
+                Promise.delay(2000)
+                    .bind(this)
+                    .then(function () {
+                        if (!this.connection.domains[this._domainName]) {
+                            var reason = "Domain '" + this._domainName +
+                                "' does not exist, nor was a domain path provided";
+                            return Promise.reject(reason);
+                        } else {
+                            console.warn("It took a second try (after delay) to get the domain from the connection");
```

Comment:
I think this is too chatty for production code. It doesn't seem like an interesting or exceptional condition. Do you want people to tell you if they see this? If not, I'd remove it.